src/ParametersHandler/YarpImplementation/src/YarpImplementation.cpp

```diff
@@ -6,33 +6,89 @@
  */
 
 #include <BipedalLocomotionControllers/ParametersHandler/YarpImplementation.h>
+#include <yarp/os/Bottle.h>
 
 #include <string>
+#include <cassert>
 
 using namespace BipedalLocomotionControllers::ParametersHandler;
 
 YarpImplementation::YarpImplementation(const yarp::os::Searchable& searchable)
 {
-    m_container.fromString(searchable.toString());
+    set(searchable);
 }
 
-std::unique_ptr<IParametersHandler<YarpImplementation>>
-YarpImplementation::getGroup(const std::string& name) const
+void YarpImplementation::set(const yarp::os::Searchable &searchable)
 {
-    auto& group = m_container.findGroup(name);
-    if (group.isNull())
-        return std::make_unique<YarpImplementation>();
+    clear();
 
-    return std::make_unique<YarpImplementation>(group);
+    yarp::os::Bottle bot;
+    bot.fromString(searchable.toString());
+
+    for (size_t i = 0; i < bot.size(); i++) { //all sublists are included in a new object
+        yarp::os::Value& bb = bot.get(i);
+
+        yarp::os::Bottle* sub = bb.asList();
+        if ((sub) && (sub->size() > 1)) {
+            std::string name = sub->get(0).toString();
+            yarp::os::Bottle* subSub = sub->get(1).asList();
+            if ((subSub) && (subSub->size() > 1))
+            {
+                m_lists.emplace(name, make_shared(*subSub));
+            }
+            else
+            {
+                m_container.add(bb);
+            }
+
+        }
+        else
+        {
+            m_container.add(bb);
+        }
+    }
+}
+
+YarpImplementation::weak_ptr YarpImplementation::getGroup(const std::string& name) const
+{
+    if (m_lists.find(name) != m_lists.end())
+    {
+        return m_lists.at(name);
+    }
+
+    return YarpImplementation::make_shared();
+}
+
+void YarpImplementation::setGroup(const std::string &name, IParametersHandler<YarpImplementation>::shared_ptr newGroup)
+{
+    auto downcastedPtr = std::dynamic_pointer_cast<YarpImplementation>(newGroup); //to access m_container
+    assert(downcastedPtr);
+    yarp::os::Bottle backup = downcastedPtr->m_container;
+    yarp::os::Bottle nameAdded;
+    nameAdded.add(yarp::os::Value(name));
+    nameAdded.append(backup);
+    downcastedPtr->m_container = nameAdded; //This is all to add the name at the beginning of the bottle
+    m_lists[name] = newGroup;
 }
 
 std::string YarpImplementation::toString() const
 {
-    return m_container.toString();
+    std::string output = m_container.toString();
+    for (auto& group : m_lists)
+    {
+        output += " (" + group.second->toString() + ")";
+    }
+    return output;
 }
 
 bool YarpImplementation::isEmpty() const
 {
     // if the toString method returns an empty string means that the handler is empty
```

Comment:
Note for me. Change this comment. Now we check if the container is null and there are no lists. A special case is when the container has a single string element. This is the case for newly created groups, where the container has only the name of the group itself.

Reply:
```suggestion
    // We check if the container is null and there are no lists. 
    // A special case is when the container has a single string element. 
    // This is the case for newly created groups, where the container has only the name of the group itself.
```